packages/crypto/src/encryption.ts

```diff
@@ -1,26 +1,32 @@
-import SHA256 from 'crypto-js/sha256';
 import AES from 'crypto-js/aes';
 import Utf8 from 'crypto-js/enc-utf8';
+import { compare, hash } from 'bcryptjs';
 
-const STRETCH_ROUNDS = 5000;
+export const SALT = '$2a$10$iLpsQfxfYc3FKZZft82qTe';
 
-// In cryptography, key stretching techniques are used to make brute-force attacks more difficult.
-// The idea is to "stretch" the key (or in this case, the password) by applying a computationally intensive
-// operation to it multiple times. This makes each attempt to guess the password more time-consuming,
-// which in turn makes brute-force attacks less feasible.
-export const repeatedHash = (message: string) => {
-  let hash = SHA256(message);
-  for (let i = 0; i < STRETCH_ROUNDS; i++) {
-    hash = SHA256(hash.toString());
-  }
-  return hash.toString();
+/**
+ * Hashes a password using bcrypt algorithm.
+ *
+ * Bcrypt is a slow hash function designed to be computationally intensive to calculate,
+ * which helps to slow down password cracking attempts. It also adds salts,
+ * which provide protection against rainbow table attacks.
+ */
+export const hashPassword = async (password: string): Promise<string> => {
+  return hash(password, SALT);
 };
 
-export const encrypt = (message: string, key: string) => {
+export const isPassword = async (attempt: string, hashedPassword: string): Promise<boolean> => {
+  return compare(attempt, hashedPassword);
+};
+
+export const encrypt = (message: string, key: string): string => {
   return AES.encrypt(message, key).toString();
 };
 
-export const decrypt = (ciphertext: string, key: string) => {
-  const decrypted = AES.decrypt(ciphertext, key);
-  return decrypted.toString(Utf8);
+// If unsuccessful, `decryptedStr` will be an empty string.
+// This function coerces that to `undefined` to make it clear to the consumer.
+export const decrypt = (ciphertext: string, key: string): string | undefined => {
+  const decryptedArr = AES.decrypt(ciphertext, key);
+  const decryptedStr = decryptedArr.toString(Utf8);
+  return decryptedStr === '' ? undefined : decryptedStr;
```

Comment:
Previously, there was an assumption that an incorrect decryption would throw. But that is not the case! It just returns `""`. So checking for that and returning `string | undefined` as a means of communicating success/failure.